devtools/src/background.js

```diff
@@ -48,8 +48,23 @@ chrome.webNavigation.onCommitted.addListener(function(details) {
   }
   let tabId = details.tabId;
   if (tabId in connections) {
-    console.log(`Registered page refresh for tab.${tabId}.`);
+    console.log(`Registered page refresh for tab.${tabId}, sending init-debug.`);
     connections[tabId].postMessage([{messageType: 'page-refresh'}]);
-    chrome.tabs.sendMessage(tabId, {messageType: 'init-debug'});
+
+    let retriesLeft = 5;
+    (function sendMessage() {
+      chrome.tabs.sendMessage(tabId, {messageType: 'init-debug'}, response => {
+        if (response) {
+          console.log(`Received response to init-debug from tab.${tabId}.`);
+          connections[tabId].postMessage(response);
+        } else if (retriesLeft--) {
+          console.log(`No response to init-debug from tab.${tabId}, retrying`);
+          // No response indicates contentScript.js didn't set up a listener yet. Retrying.
+          setTimeout(sendMessage, 200);
```

Comment:
Arbitrary fixed timeouts make me a bit sad, but I guess it's the best way to handle this...

Reply:
Hmm, I'll see how to handle this better - I think you're right this is ugly/wrong-ish.

Reply:
Fixed, thanks for pushing back!